src/main/java/org/apache/xml/security/encryption/AgreementMethod.java

```diff
@@ -88,6 +91,22 @@ public interface AgreementMethod {
      */
     void setKANonce(byte[] kanonce);
 
+
+    /**
+     * Returns KeyDerivationMethod information used in the <code>AgreementMethod</code>.
+     * @return The KeyDerivationMethod information regarding the <code>AgreementMethod</code>.
+     */
+    KeyDerivationMethod getKeyDerivationMethod() throws XMLSecurityException;
```

Comment:
By adding new methods to an interface, this means applications must use the version of this API in which the methods are added (they can't expect to use older versions w/o the methods). I guess that's ok, but not sure what our policy is on that - does this suggest we need a new major version when this is released,  @coheigea ?

Reply:
I am OK with it as I doubt anyone is using it. Maybe we restrict this change to 4.0.x though and create a new 3.1.0 for Java 8.